### PR TITLE
Add Pumping Duration and Create from Quick Timer

### PR DESCRIPTION
**Pull request by johnmerlie**
This PR covers adding start/end times and duration (as exist on `feeding`) to `pumping`, as well as adding the ability to create a `pumping` entry from a quick timer as discussed in #573

### CODE REVIEW

**Comment by cdubz** [comment]
@johnmerlie this is all looking about right. I'd like to get the _existing_ tests updated before I do a thorough review here.

It looks like a big chunk of the current fails are just related to the now outdated fixtures. There is only one pumping entry in the fixtures:

https://github.com/johnmerlie/babybuddy/blob/1fef726b9b127ed7349bbcba40987da6e420ef29/babybuddy/fixtures/tests.json#L24-L34

You should be able to essentially update that by hand and then get a better idea of what tests actually do need to be updated.

**Comment by johnmerlie** [comment]
@cdubz Thanks for the guidance. I've gone back and updated all the test data/cases, including adding an additional pumping entry (to double check sorting changes). 

Looks like everything is passing locally when I run gulp test, and checks are passing. 

The one thing I wasn't sure about is whether there's a good way to test the migration on existing data - from what i can tell it looks like all the fake data for testing is generated _after_ the migration command, so it's naive to old models. If that's expected and explicit testing of the data migrations isn't explicitly tested, then this should be about ready.

Let me know what you think

File: core/migrations/0029_alter_pumping_options_remove_pumping_time_and_more.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Generated by Django 4.2.2 on 2023-06-16 20:06
+
+from django.db import migrations, models
+import django.utils.timezone
+
+
+def set_pumping_start_end_values(apps, schema_editor):
+    # The model must be imported to ensure its overridden `save` method is run.
+    from core.models import Pumping
```
Comment (cdubz) [inline core/migrations/0029_alter_pumping_options_remove_pumping_time_and_more.py:9]: To ensure you have access to the old `time` field you'll need to get `Pumping` from `apps` -- e.g., `Pumping = apps.get_model("core", "Pumping")`. The `apps` variable represents the model _before_ the migration.

File: api/views.py [hunk 2]
```diff
@@ -82,8 +82,8 @@ class PumpingViewSet(viewsets.ModelViewSet):
     queryset = models.Pumping.objects.all()
     serializer_class = serializers.PumpingSerializer
     filterset_class = filters.PumpingFilter
-    ordering_fields = ("amount", "time")
-    ordering = "-time"
+    ordering_fields = ("amount", "duration", "end", "start")
+    ordering = "-start"
```
Comment (cdubz) [inline api/views.py:86]: For now lets follow the same default ordering as `Feeding` and set this to `-end`.

File: core/templates/core/pumping_list.html
Comment (cdubz) [inline core/templates/core/pumping_list.html:1]: Rename the column header here `Date` just to keep this in parity with the Feeding list.

**Review by cdubz — CHANGES_REQUESTED** [review]
Mostly looks good! Just a few small CRs here.

> The one thing I wasn't sure about is whether there's a good way to test the migration on existing data - from what i can tell it looks like all the fake data for testing is generated after the migration command, so it's naive to old models. If that's expected and explicit testing of the data migrations isn't explicitly tested, then this should be about ready.

I have never got around to building in any sort of migration testing. I probably _should_ but for now it's just manual. So no worries on that front.

File: core/migrations/0029_alter_pumping_options_remove_pumping_time_and_more.py (hunk 1 above)
Comment (johnmerlie) [reply core/migrations/0029_alter_pumping_options_remove_pumping_time_and_more.py]: fixed with [b7b28926c164a6cae38ff750fdbde79f1b2fa3e6](https://github.com/babybuddy/babybuddy/pull/665/commits/b7b28926c164a6cae38ff750fdbde79f1b2fa3e6)

File: api/views.py (hunk 2 above)
Comment (johnmerlie) [reply api/views.py]: fixed with [b8a1e39987cfcad5a7d59cb60b0e153812dc2b94](https://github.com/babybuddy/babybuddy/pull/665/commits/b8a1e39987cfcad5a7d59cb60b0e153812dc2b94)

File: core/templates/core/pumping_list.html
Comment (johnmerlie) [reply core/templates/core/pumping_list.html:1]: fix for [2485f0309225f61f04058c1f2c7b19267c9bd000](https://github.com/babybuddy/babybuddy/pull/665/commits/2485f0309225f61f04058c1f2c7b19267c9bd000)

**Comment by johnmerlie** [comment]
@cdubz  Should all be fixed, I'll let you take a look and mark as resolved if you agree

> I have never got around to building in any sort of migration testing. I probably should but for now it's just manual. So no worries on that front.

No worries, just wanted to make sure I wasn't missing anything!

**Review by cdubz — APPROVED** [review]
Looks good. Thanks for all your work on this!!

**Comment by johnmerlie** [comment]
> Looks good. Thanks for all your work on this!!

@cdubz babybuddy has been a huge help with our (almost) 7 week old, so it feels like the least I can do!

**Pull request merged**